cmd/ingester/app/processor/span_processor.go

```diff
@@ -31,6 +31,13 @@ type SpanProcessor interface {
 	io.Closer
 }
 
+// ParallelSpanProcessor processes kafka spans in parallel.
+type ParallelSpanProcessor interface {
+	Process(input Message)
+	Errors() <-chan error
```

Comment:
I am not a fan of this design. It provides back a stream of anonymous errors, not connected to the original messages. In general, I dislike using channels in the API. You could use a callback mechanism instead, which would link the errors to messages, and you can still convert it to channel processing on the caller side if necessary, without creating cross-component channel dependencies.
```go
type ParallelSpanProcessor interface {
    Process(input Message, onErr func(Message, error))
     io.Closer
```